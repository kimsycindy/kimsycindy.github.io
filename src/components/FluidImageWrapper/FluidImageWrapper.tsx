import React from 'react'

interface Props {
  width?: number
  height?: number
  className?: string
}

const FluidImageWrapper: React.FC<Props> = ({
  width,
  height,
  className,
  children,
}) => (
  <div
    className={className}
    style={{
      width,
      height,
      maxWidth: width,
      maxHeight: height,
      marginBottom: 20,
    }}
  >
    {children}
  </div>
)

export default FluidImageWrapper
