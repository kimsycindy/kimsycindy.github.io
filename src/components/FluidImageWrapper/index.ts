import FluidImageWrapper from './FluidImageWrapper'

export default FluidImageWrapper
