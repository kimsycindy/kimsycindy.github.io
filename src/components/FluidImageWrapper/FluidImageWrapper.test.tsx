import React from 'react'
import renderer from 'react-test-renderer'

import Img from 'gatsby-image'
import FluidImageWrapper from './FluidImageWrapper'

describe('FluidImageWrapper', () => {
  it('renders correctly', () => {
    const fluid = {
      aspectRatio: 1,
      src: 'https://cindy.kim',
      srcSet: 'elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w',
      sizes: '(max-width: 600px) 480px, 800px',
    }

    const tree = renderer
      .create(
        <FluidImageWrapper
          width={500}
          height={500}
          className="fluidImageWrapper-mock"
        >
          <Img fluid={fluid} />
        </FluidImageWrapper>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
