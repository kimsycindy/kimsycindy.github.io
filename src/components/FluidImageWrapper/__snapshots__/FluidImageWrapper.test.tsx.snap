// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FluidImageWrapper renders correctly 1`] = `
<div
  className="fluidImageWrapper-mock"
  style={
    Object {
      "marginBottom": 20,
      "maxHeight": 500,
      "maxWidth": 500,
    }
  }
>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      aria-hidden={true}
      style={
        Object {
          "paddingBottom": "100%",
          "width": "100%",
        }
      }
    />
    <picture>
      <source
        sizes="(max-width: 600px) 480px, 800px"
        srcSet="elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w"
      />
      <img
        alt=""
        loading="lazy"
        onLoad={[Function]}
        sizes="(max-width: 600px) 480px, 800px"
        src="https://cindy.kim"
        srcSet="elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w\\" sizes=\\"(max-width: 600px) 480px, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 600px) 480px, 800px\\" srcset=\\"elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w\\" src=\\"https://cindy.kim\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
</div>
`;
